storage/innobase/srv/srv0start.cc

```diff
@@ -3104,6 +3105,8 @@ innodb_shutdown()
 		srv_undo_sources = false;
 	}
 
+	my_sd_extend_timeout(INNODB_EXTEND_TIMEOUT_INTERVAL);
```

Comment:
This call could be moved to the start of `logs_empty_and_mark_files_at_shutdown()` where all the action is.
I do not see any need to modify this file in this patch.